Xamarin.Forms.Core/BorderElement.cs

```diff
@@ -1,14 +1,86 @@
+
 		public static readonly BindableProperty BorderColorProperty =
 			BindableProperty.Create("BorderColor", typeof(Color), typeof(IBorderElement), Color.Default,
 									propertyChanged: OnBorderColorPropertyChanged);
 
+		public static readonly BindableProperty BorderWidthProperty = BindableProperty.Create("BorderWidth", typeof(double), typeof(IBorderElement), -1d);
+
+		[Obsolete("BorderRadiusProperty is obsolete as of 2.5.0. Please use CornerRadius instead.")]
+		public static readonly BindableProperty BorderRadiusProperty = BindableProperty.Create("BorderRadius", typeof(int), typeof(IBorderElement), defaultValue: DefaultBorderRadius,
+			propertyChanged: BorderRadiusPropertyChanged);
+
+		public static readonly BindableProperty CornerRadiusProperty = BindableProperty.Create("CornerRadius", typeof(int), typeof(IBorderElement), defaultValue: DefaultCornerRadius,
+			propertyChanged: CornerRadiusPropertyChanged);
+
 		static void OnBorderColorPropertyChanged(BindableObject bindable, object oldValue, object newValue)
 		{
 			((IBorderElement)bindable).OnBorderColorPropertyChanged((Color)oldValue, (Color)newValue);
 		}
+
+
+		static void BorderRadiusPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
+		{
+			if (newvalue == oldvalue)
+				return;
+
+			var borderElement = (IBorderElement)bindable;
+			var borderOrCornerRadius = (bindable as IBorderOrCornerRadius);
+
+			var val = (int)newvalue;
+			if (val == DefaultBorderRadius && (!borderOrCornerRadius?.cornerOrBorderRadiusSetting ?? true))
+				val = DefaultCornerRadius;
+
+			var oldVal = (int)bindable.GetValue(Button.CornerRadiusProperty);
+
+			if (oldVal == val)
+				return;
+
+			if (borderOrCornerRadius != null)
+				borderOrCornerRadius.cornerOrBorderRadiusSetting = true;
+
+			bindable.SetValue(Button.CornerRadiusProperty, val);
+
+			if (borderOrCornerRadius != null)
+				borderOrCornerRadius.cornerOrBorderRadiusSetting = false;
+		}
+
+		static void CornerRadiusPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
+		{
+			if (newvalue == oldvalue)
+				return;
+
+			var borderElement = (IBorderElement)bindable;
+			var borderOrCornerRadius = (bindable as IBorderOrCornerRadius);
+
+			var val = (int)newvalue;
+
+			if (val == DefaultCornerRadius && (!borderOrCornerRadius?.cornerOrBorderRadiusSetting ?? true))
+				val = DefaultBorderRadius;
+
+#pragma warning disable 0618 // retain until BorderRadiusProperty removed
+			var oldVal = (int)bindable.GetValue(Button.BorderRadiusProperty);
+#pragma warning restore
+
+			if (oldVal == val)
+				return;
+
+#pragma warning disable 0618 // retain until BorderRadiusProperty removed
+			if (borderOrCornerRadius != null)
+				borderOrCornerRadius.cornerOrBorderRadiusSetting = true;
+
+			bindable.SetValue(BorderElement.BorderRadiusProperty, val);
+
+			if (borderOrCornerRadius != null)
+				borderOrCornerRadius.cornerOrBorderRadiusSetting = false;
+#pragma warning restore
+		}
```

Comment:
I trust that this code was copied over the `Button` class and keep working as is

Reply:
yea I just copied this code
https://github.com/xamarin/Xamarin.Forms/pull/4328/files#diff-d8507c43c20cade63aeeaa2507bd1509L265

But on second assessment the only point of this code is to harmonize Corner Radius and Border Radius on the Button.

Other than that the property changed handlers serve no purpose so I'm just going to delete all this code from the Border Element so as not to pollute everything else